tests/load/sources/sql_database/sql_source.py

```diff
@@ -0,0 +1,373 @@
+                channel_ids.extend(result.scalars())
+        info["row_count"] += n
+        info["ids"] += channel_ids
+        return channel_ids
+
+    def fake_messages(self, n: int = 9402) -> List[int]:
+        user_ids = self.table_infos["app_user"]["ids"]
+        channel_ids = self.table_infos["chat_channel"]["ids"]
+        _text = mimesis.Text()
+        choice = mimesis.Choice()
+        table = self.metadata.tables[f"{self.schema}.chat_message"]
+        message_ids: List[int] = []
+        info = self.table_infos.setdefault(
+            "chat_message",
+            dict(row_count=0, ids=[], created_at=IncrementingDate(), is_view=False),
+        )
+        dt = info["created_at"]
+        for chunk in chunks(range(n), 5000):
+            rows = [
+                dict(
+                    content=_text.random.choice(_text.extract(["questions"])),
+                    user_id=choice(user_ids),
+                    channel_id=choice(channel_ids),
+                    created_at=next(dt),
+                    updated_at=next(dt),
+                )
+                for i in chunk
+            ]
+            with self.engine.begin() as conn:
+                result = conn.execute(table.insert().values(rows).returning(table.c.id))
+                message_ids.extend(result.scalars())
+        info["row_count"] += len(message_ids)
+        info["ids"].extend(message_ids)
+        # View is the same number of rows as the table
+        view_info = deepcopy(info)
+        view_info["is_view"] = True
+        view_info = self.table_infos.setdefault("chat_message_view", view_info)
+        view_info["row_count"] = info["row_count"]
+        view_info["ids"] = info["ids"]
+        return message_ids
+
+    def _fake_precision_data(self, table_name: str, n: int = 100, null_n: int = 0) -> None:
+        table = self.metadata.tables[f"{self.schema}.{table_name}"]
+        self.table_infos.setdefault(table_name, dict(row_count=n + null_n, is_view=False))  # type: ignore[call-overload]
+        rows = [
+            dict(
+                int_col=random.randrange(-2147483648, 2147483647),
+                bigint_col=random.randrange(-9223372036854775808, 9223372036854775807),
+                smallint_col=random.randrange(-32768, 32767),
+                numeric_col=random.randrange(-9999999999, 9999999999) / 100,
+                numeric_default_col=random.randrange(-9999999999, 9999999999) / 100,
+                string_col=mimesis.Text().word()[:10],
+                string_default_col=mimesis.Text().word(),
+                datetime_tz_col=mimesis.Datetime().datetime(timezone="UTC"),
+                datetime_ntz_col=mimesis.Datetime().datetime(),  # no timezone
+                date_col=mimesis.Datetime().date(),
+                time_col=mimesis.Datetime().time(),
+                float_col=random.random(),
+                json_col='{"data": [1, 2, 3]}',  # NOTE: can we do this?
```

Comment:
fyi: this is change from an actual object to a serialized json and this fixes a number of problems in the tests. I think this is fine.

Reply:
a string is a valid JSON object so now we test if we can store a string. not sure those tests are meaningful but I think it is good enough